server/src/crm/crm.service.ts

```diff
@@ -66,340 +116,46 @@ export class CrmService {
-            zlib.gunzip(body, (err, dezipped) => {
-              parseError(dezipped);
-            });
-          }
-          else {
-            parseError(body);
-
-          }
-        }
-        else if (response.statusCode === 200 || response.statusCode === 201) {
-          const parseResponse = jsonText => {
-            const json_string = jsonText.toString('utf-8');
-            var result = JSON.parse(json_string, this._dateReviver);
-            resolve(result);
-          };
-
-          if (encoding && encoding.indexOf('gzip') >= 0) {
-            zlib.gunzip(body, (err, dezipped) => {
-              parseResponse(dezipped);
-            });
-          }
-          else {
-            parseResponse(body);
-          }
-        }
-        else if (response.statusCode === 204 || response.statusCode === 1223) {
-          const uri = response.headers["OData-EntityId"];
-          if (uri) {
-            // create request - server sends new id
-            const regExp = /\(([^)]+)\)/;
-            const matches = regExp.exec(uri);
-            const newEntityId = matches[1];
-            resolve(newEntityId);
-          }
-          else {
-            // other type of request - no response
-            resolve();
-          }
-        }
-        else {
-          resolve();
-        }
-      });
-    })
-  }
-
-  async _update(entitySetName, guid, data, headers) {
-    var query = entitySetName + "(" + guid + ")";
-    return this._sendPatchRequest(query, data, headers);
-  }
-
-  async _sendPatchRequest(query, data, headers) {
-    //  get token
-    const JWToken = await this.ADAL.acquireToken();
-    const options = {
-      url: `${this.host + query}`,
-      headers: {
-        'Accept-Encoding': 'gzip, deflate',
-        'Content-Type': 'application/json; charset=utf-8',
-        Authorization: `Bearer ${JWToken}`,
-        'OData-MaxVersion': '4.0',
-        'OData-Version': '4.0',
-        Accept: 'application/json',
-        Prefer: 'odata.include-annotations="*"',
-        ...headers
-      },
-      body: JSON.stringify(data),
-      encoding: null,
-    };
-
-    return new Promise((resolve, reject) => {
-      Request.patch(options, (error, response, body) => {
-        const encoding = response.headers['content-encoding'];
-
-        if (error || response.statusCode != 204) {
-          const parseError = jsonText => {
-            const json_string = jsonText.toString('utf-8');
-            var result = JSON.parse(json_string, this._dateReviver);
-            var err = this._parseErrorMessage(result);
-            reject(err);
-          };
-          if (encoding && encoding.indexOf('gzip') >= 0) {
-            zlib.gunzip(body, (err, dezipped) => {
-              parseError(dezipped);
-            });
-          }
-          else {
-            parseError(body);
-
-          }
-        }
-        else resolve(body);
-      })
-    });
-  }
-
-  async _sendDeleteRequest(query, headers) {
```

Comment:
WARNING: If we eventually merge Search and Portal APIs, make sure this deletion doesn't take precedent. 

Applicant Portal API depends on this method:
https://github.com/NYCPlanning/labs-applicant-portal/blob/develop/server/src/crm/crm.service.ts#L410

Reply:
Maybe we should preserve  this function and migrate it to superagent as well?

Reply:
will do in follow up